commit 47f1861c0d78a7091ecd47e1f165612ac810d17a
Author: Matthieu CERDA <matthieu.cerda@normation.com>
Date:   Wed Jun 25 19:07:23 2014 +0200

    Redmine#6189: Store the hardcoded bootstrap file in a variable instead of using it directly to prevent old GCCs from failing due to fprintf usage as a macro

diff --git a/libpromises/bootstrap.c b/libpromises/bootstrap.c
index f28d878..e126e8d 100644
--- a/libpromises/bootstrap.c
+++ b/libpromises/bootstrap.c
@@ -284,6 +284,11 @@ bool MasterfileExists(const char *masterdir)
 
 bool WriteBuiltinFailsafePolicyToPath(const char *filename)
 {
+    // The bootstrap.inc file is generated by "make bootstrap-inc"
+    char * bootstrap_content =
+#include "bootstrap.inc"
+;
+
     Log(LOG_LEVEL_INFO, "Writing built-in failsafe policy to '%s'", filename);
 
     FILE *fout = fopen(filename, "w");
@@ -293,10 +298,7 @@ bool WriteBuiltinFailsafePolicyToPath(const char *filename)
         return false;
     }
 
-    fprintf(fout,
-            // The bootstrap.inc file is generated by "make bootstrap-inc"
-#include "bootstrap.inc"
-    );
+    fprintf(fout, bootstrap_content);
     fclose(fout);
 
     if (chmod(filename, S_IRUSR | S_IWUSR) == -1)
